lib/oidcstrategy.js

```diff
@@ -47,6 +47,10 @@ const Metadata = require('./metadata').Metadata;
 // global variable definitions
 const log = new Log('AzureAD: OIDC Passport Strategy');
 
+// save the state used in authentication request here, so we can compare it with
+// the returning state from the response
+var saved_state = null;
```

Comment:
Correct me if I'm wrong, but using a global variable like this means this will break when you have more than one OAuth process running concurrently. I think it needs to be stored per user/session. This is one of the ways that passport-oauth2 handles it: https://github.com/jaredhanson/passport-oauth2/blob/master/lib/state/session.js